app/controllers/rentals_controller.rb

```diff
@@ -76,11 +76,16 @@ def create
 
     @start_date = params['start_date'] || Time.zone.today
 
-    if @rental.save
+    if @rental.create_reservation && @rental.save
```

Comment:
We do, but it's not checking that `create_reservation` works correctly. If `create_reservation` doesn't work and returns false, it will still attempt to perform `@rental.save` thus causing the error page to appear. Not sure of the best way besides this to verify that the hook runs correctly.